### ISSUE TITLE
web: illegal state exception when compiling  

### ISSUE THREAD

**Issue by eymar** [JetBrains/compose-jb#1306]
https://kotlinlang.slack.com/archives/C01F2HV7868/p1631810459103900

```
java.lang.IllegalStateException: ComposableLambdaImpl.invoke() not found CALL 'public abstract fun invoke (p1: P1 of kotlin.Function1): R of kotlin.Function1 [operator] declared in kotlin.Function1' type=kotlin.Unit origin=INVOKE
  $this: GET_VAR 'val setModalState: kotlin.Function1<kotlin.Function2<androidx.compose.runtime.Composer, kotlin.Int, kotlin.Unit>, kotlin.Unit> [val] declared in composables.alertsComposable$composable' type=kotlin.Function1<kotlin.Function2<androidx.compose.runtime.Composer, kotlin.Int, kotlin.Unit>, kotlin.Unit> origin=VARIABLE_AS_FUNCTION
  p1: CALL 'public final fun composableLambdaInstance (key: kotlin.Int, tracked: kotlin.Boolean, block: kotlin.Any): androidx.compose.runtime.internal.ComposableLambda declared in androidx.compose.runtime.internal' type=androidx.compose.runtime.internal.ComposableLambda origin=null
    key: CONST Int type=kotlin.Int value=-985535590
 ```

Full stack trace: https://pastebin.com/rgfings1

breaks when compiling this code:

```kotlin
val (modalState, setModalState) = remember { mutableStateOf<(@Composable () -> Unit)>({ loading() }) }
```

Workaround:
```kotlin
var modalState by remember { mutableStateOf<(@Composable () -> Unit)>({ loading() }) }

//then later:

modalState = { SomeOtherComposable() }
```

**Comment by eymar** [comment]
https://github.com/JetBrains/compose-jb/issues/1436 - another related issue, that prevents the desired syntax to work as expected.

**Comment by eymar** [comment]
the compilation issue is fixed in 1.0.0 - closing this issue

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5229">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
